src/geode/geometry/dynamic_nn_search.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (c) 2019 - 2024 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/geometry/dynamic_nn_search.hpp>
+
+#include <numeric>
+
+#include <absl/algorithm/container.h>
+
+#include <async++.h>
+
+#include <nanoflann.hpp>
+
+#include <geode/basic/logger.hpp>
+#include <geode/basic/pimpl_impl.hpp>
+
+namespace geode
+{
+    template < index_t dimension >
+    class DynamicNNSearch< dimension >::Impl
+    {
+    public:
+        explicit Impl( std::vector< Point< dimension > > points )
+            : cloud_{ std::move( points ) },
+              dynamic_nn_tree_{ dimension, cloud_ }
+        {
+        }
+
+        const Point< dimension >& point( const index_t index ) const
+        {
+            return cloud_.points.at( index );
+        }
+
+        index_t nb_points() const
+        {
+            return cloud_.kdtree_get_point_count();
+        }
+
+        std::vector< index_t > radius_neighbors(
+            const Point< dimension >& point,
+            const double threshold_distance ) const
+        {
+            std::vector< nanoflann::ResultItem< index_t, double > > results;
+            nanoflann::RadiusResultSet< double, index_t > resultSet(
+                threshold_distance * threshold_distance, results );
+            dynamic_nn_tree_.findNeighbors( resultSet, &copy( point )[0] );
+            std::sort(
+                results.begin(), results.end(), nanoflann::IndexDist_Sorter() );
```

Comment:
```suggestion
            absl::c_sort( results, nanoflann::IndexDist_Sorter() );
```
abseil provides helpers for every stl algorithms